catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/suggestion/DmsCoordinateProcessorTest.java

```diff
@@ -0,0 +1,188 @@
+  }
+
+  @Test
+  public void testDmsStringNoDirection() {
+    assertSuggestionDoesNotExist("28°56\'26\" 117°38\'11\"");
+  }
+
+  @Test
+  public void testDmsStringLatDegreesTooLarge() {
+    assertSuggestionDoesNotExist("91°56\'26\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLonDegreesTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 181°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLatMinutesTooLarge() {
+    assertSuggestionDoesNotExist("28°60\'26\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLonMinutesTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 117°60\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLatSecondsTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'60\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLonSecondsTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 117°38\'60\"W");
+  }
+
+  @Test
+  public void testDmsStringConvertedLatTooLarge() {
+    assertSuggestionDoesNotExist("90°00\'01\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringConvertedLonTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 180°00\'01\"W");
+  }
+
+  @Test
+  public void testDmsStringMultipleCoordinates() {
+    assertSuggestion(
+        "28°56\'26\"N 117°38\'11.64564\"W 28;56;26S 117°38\'11\"QQW%WWEQW@!!!!!! 28°56\'26\"N 117°38\'11\"E",
+        "DMS: [ 28°56'26\"N 117°38'11.646\"W ] [ 28°56'26\"S 117°38'11\"W ] [ 28°56'26\"N 117°38'11\"E ]",
+        ImmutableList.of(
+            new LatLon(28.940555, -117.636568),
+            new LatLon(-28.940555, -117.636388),
+            new LatLon(28.940555, 117.636388)));
+  }
+
+  private void assertSuggestionDoesNotExist(String query) {
+    List<Suggestion> list = new LinkedList<>();
+    processor.enhanceResults(list, query);
+    assertThat(list, is(empty()));
+  }
+
+  private void assertSuggestion(String query, String expectedName, List<LatLon> expectedGeo) {
+    List<Suggestion> list = new LinkedList<>();
+    processor.enhanceResults(list, query);
+    assertThat(list, is(not(empty())));
+    assertThat(list, hasSize(1));
+
+    LiteralSuggestion literalSuggestion = (LiteralSuggestion) list.get(0);
+    assertThat(literalSuggestion.getId(), is("LITERAL-DMS"));
+    assertThat(literalSuggestion.getName(), is(expectedName));
+    assertThat(literalSuggestion.hasGeo(), is(true));
+
+    List<LatLon> actualGeo = literalSuggestion.getGeo();
+    List<LatLon> diff = new ArrayList<>();
+    assertThat(
+        "Actual geo and expected geo are the same size",
+        actualGeo.size(),
+        is(equalTo(expectedGeo.size())));
+    for (int i = 0; i < expectedGeo.size(); i++) {
+      LatLon actual = actualGeo.get(i);
+      LatLon expected = expectedGeo.get(i);
+      Double latDiff = actual.getLat() - expected.getLat();
+      Double lonDiff = actual.getLon() - expected.getLon();
+
+      if (Math.abs(latDiff) > 0.00001 || Math.abs(lonDiff) > 0.00001) {
```

Comment:
✏️ Could use the `closeTo()` matcher instead:
```java
assertThat("The latitudes are not equal", actual.getLat(), is(closeTo(expected.getLat(), 0.00001)));
assertThat("The longitudes are not equal", actual.getLon(), is(closeTo(expected.getLon(), 0.00001)));
```
As this is written right now, if either the latitudes or the longitudes don't match, no useful information will be printed to aid in determining why the test failed.

Reply:
I have the assertion after the for loop which I think would be better to fail on. It will show which coordinates were not as expected for the whole geo, instead of failing on one.

Reply:
Ah, so this will print the contents of `diff` when the test fails. In that case, the message in the assertion should say that some geos were _not_ matched.

Reply:
Yeah, I ended up changing it to that based on your other comment about the geo size